### ISSUE TITLE
Replication job doesn't timeout

### ISSUE THREAD

**Issue by angelbarrera92** [goharbor/harbor#13799]
Ciao team!

In a scenario where the connectivity to the two registries involved in a replication job fails, 
a scheduled replication job starts but remains in the pending state without timing out. This makes it difficult to understand what is going on.

I think it could be a good idea to enable the configuration of a timeout in the replication jobs. What do you think?

**Comment by ywk253100** [comment]
Could you check whether the job is running in `jobservice`? Maybe this is just the case that the status is out of syncing

**Comment by ChristianCiach** [comment]
Yes, we see the same issue. Sometimes we have to manually kill hundreds of "running" jobs.

I really think that there never should be more than one running job for the same replication. A job should be skipped of there is a previous job of the same kind still running.

**Comment by ywk253100** [comment]
@ChristianCiach You can configure the `max_job_workers` to control the running jobs in the same time, the default value is `10`.
BTW, one replication job handles the artifacts of a single repository, so when you trigger a replication policy which covers several repositories, there will be several replication jobs

**Comment by ChristianCiach** [comment]
@ywk253100 No, that doesn't work. We have `max_job_workers` configured to "2", but when replication jobs hang for some reasons (like connection issues), the next run will be started even if the previous one never terminates. Because our single pull based job is scheduled to run every 5 minutes, this means there will be 12 jobs showing as "running" in the gui every hour. These pile up until the connection issue is resolved or they are terminated manually. Thats the reason why we sometimes have to kill hundreds of "running" jobs manually.

**Comment by steven-zou** [comment]
add a timeout to the job, this is somehow a new requirement.

**Comment by ChristianCiach** [comment]
A timeout will not solve the core issue here, being that the same replication job is run multiple times in parallel. Sure, a timeout will reduce that *amount* of instances of the same job running in parallel, but there will still be multiple instances of the same job, just fewer of them. This is a workaround at best, but not a solution.

**Comment by steven-zou** [comment]
> A timeout will not solve the core issue here, being that the same replication job is run multiple times in parallel. Sure, a timeout will reduce that _amount_ of instances of the same job running in parallel, but there will still be multiple instances of the same job, just fewer of them. This is a workaround at best, but not a solution.

We're also considering supporting unique running jobs not just unique queuing jobs in the future.

**Comment by defreng** [comment]
This is actually a serious issue for production deployments of Harbor - because if the issue got really bad, it's almost impossible to recover.

There is no (known to me) way to mass-kill running jobs. In our case, Harbor actually ended up in an endless crashloop together with Redis and the DB.

**Comment by 0xj0el** [comment]
Any update here
